backends/es6/indexed_publication.go

```diff
@@ -124,12 +125,15 @@ func NewIndexedPublication(p *models.Publication) *indexedPublication {
 
 	for _, supervisor := range p.Supervisor {
 		ip.Contributor = append(ip.Contributor, supervisor.Name())
+		if supervisor.PersonID != "" {
+			ip.SupervisorID = append(ip.SupervisorID, supervisor.PersonID)
+		}
 	}
+	ip.SupervisorID = lo.Uniq(ip.SupervisorID)
```

Comment:
this is a bit brute force, Uniq makes a lookup map etc
can simply be:
```
if !slices.Contains(ip.SupervisorID, supervisor.PersonID) { append(...) }
```

Reply:
Changed this. Did the same in other places that I didn't touch for this PR.